app/src/androidTest/java/ch/epfl/sdp/peakar/points/TopographyAsyncTest.java

```diff
@@ -101,166 +102,127 @@ public void getBoundingBoxWestLongTest() {
+        // Check if the points are filtered correctly
+        Assert.assertEquals(new HashSet<>(lineOfSight.getVisiblePoints(pointsToCheck)), new HashSet<>(visiblePoints));
     }
 
-    /*
-    Used for debug purposes, not for testing
+    /**
+     * Checks if the altitude is calculated correctly and that the elevation map gets udated correctly
      */
-    public void debugTest() throws ExecutionException, InterruptedException {
+    @Test
+    public void G_getAltitudeAndIndexesTest() throws InterruptedException {
 
-        // setting location near everest peak
         UserPoint userPoint = UserPoint.getInstance(mContext);
-        userPoint.setLocation(46.52301918715015, 6.56573360554917,400, 0);
-
-        new DownloadTopographyTask(mContext){
-            @Override
-            public void onResponseReceived(Pair<int[][], Double> topography) {
-                super.onResponseReceived(topography);
-                topographyPair = topography;
-            }
-        }.execute(userPoint).get();
+        userPoint.setLocation(GPSTracker.DEFAULT_LAT, GPSTracker.DEFAULT_LON,GPSTracker.DEFAULT_ALT, GPSTracker.DEFAULT_ACC);
 
-        LineOfSight lineOfSight = new LineOfSight(topographyPair, userPoint, mContext);
+        ElevationMap elevationMap = new ElevationMap(topographyPair, userPoint, mContext);
 
-        // PoiPoints to check
-        List<POIPoint> pointsToCheck = new ArrayList<>();
-        POIPoint point1 = new POIPoint(new GeoPoint(46.19544415848338, 6.6072792585438656, 2244));
-        point1.setName("point1");
-        pointsToCheck.add(point1);
+        // check the altitude around the Everest Peak using coordinates
+        Assert.assertEquals(8849, elevationMap.getAltitudeAtLocation(GPSTracker.DEFAULT_LAT, GPSTracker.DEFAULT_LON), 200);
+        // check the altitude around the Everest Peak using indexes
+        Pair<Integer, Integer> indexes = elevationMap.getIndexesFromCoordinates(GPSTracker.DEFAULT_LAT, GPSTracker.DEFAULT_LON);
+        // check the indexes around the Everest Peak
+        Assert.assertEquals(new Pair<>(539, 610), indexes);
+        Assert.assertNotNull(indexes);
+        Assert.assertNotNull(indexes.first);
+        Assert.assertNotNull(indexes.second);
+        Assert.assertEquals(8849, elevationMap.getAltitudeAtLocation(indexes.first, indexes.second), 200);
 
-        Map<POIPoint, Boolean> filtered = lineOfSight.getVisiblePointsLabeled(pointsToCheck);
+        int[][] oldTopographyMap = topographyPair.first;
 
-        Log.d("DEBUG", filtered.toString());
+        // set location near the Mont Blanc
+        userPoint.setLocation(45.802537, 6.850328, 0, 0);
+        elevationMap.updateElevationMatrix();
 
-        Assert.assertTrue(true);
+        //Wait for the map to be updated
+        int counter=0;
+        while(elevationMap.getTopographyMap()==oldTopographyMap && counter<50){
+            Thread.sleep(1000);
```

Comment:
In the testing constants class, we have a constant called `LONG_SLEEP_TIME` (or something like this) that you could use instead of a hard-coded constant.

Reply:
I actually didn't code this part of the test. @magangloff do you agree with @giommok suggestion?

Reply:
Yes, you can simply use THREAD_SLEEP_1S or LONG_SLEEP_TIME from the constants, the hard-coded constant was before we introduced the testing constants